### PR TITLE
Fix links to make them all accessible

### PR DESCRIPTION
**Pull request by SajidAlamQB**
Signed-off-by: SajidAlamQB <90610031+SajidAlamQB@users.noreply.github.com>

## Description

Some of our links are still failing basic accessibility which is fine for now but we should fix them someday. We can adjust some just as we work on pages but should put some time aside to go through and fix them all. 

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1690"></a>

### CODE REVIEW

File: docs/source/extend_kedro/common_use_cases.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ You can use [DataSets](/kedro.extras.datasets) to interface with various differe
 
 ## Use Case 3: How to add or modify CLI commands
 
-If you want to customise a built-in Kedro command, such as `kedro run`, for a specific project, add a `cli.py` file that defines a custom `run()` function. You should add the `cli.py` file at the same level as `settings.py`, which is usually the `src/PROJECT_NAME` directory. A template for the `cli.py` file can be found [here](../development/commands_reference.md#customise-or-override-project-specific-kedro-commands).
+If you want to customise a built-in Kedro command, such as `kedro run`, for a specific project, add a `cli.py` file that defines a custom `run()` function. You should add the `cli.py` file at the same level as `settings.py`, which is usually the `src/PROJECT_NAME` directory. Find the [template for the `cli.py` file](../development/commands_reference.md#customise-or-override-project-specific-kedro-commands).
```
Comment (deepyaman) [inline docs/source/extend_kedro/common_use_cases.md:19]:
```suggestion
See [the template for the `cli.py` file](../development/commands_reference.md#customise-or-override-project-specific-kedro-commands).
```
Make it less of a command. :)

**Review by MerelTheisenQB — APPROVED** [review]
Thank you, the links have much better descriptions now! 😄

**Review by deepyaman — CHANGES_REQUESTED** [review]
Some questions and recommendations.

File: docs/source/development/commands_reference.md [hunk 2]
```diff
@@ -303,7 +303,7 @@ Call the `run()` method of the `KedroSession` defined in `kedro.framework.sessio
 kedro run
 ```
 
-`KedroContext` can be extended in `run.py` (`src/<package_name>/run.py`). In order to use the extended `KedroContext` you need to set `context_path` in [`pyproject.toml`](../faq/architecture_overview.md#kedro-project) configuration file.
+`KedroContext` can be extended in `run.py` (`src/<package_name>/run.py`). In order to use the extended `KedroContext` you need to set `context_path` in [`pyproject.toml` configuration file](../faq/architecture_overview.md#kedro-project).
```
[the remark this answers is not in the document]
Comment (yetudada) [reply docs/source/development/commands_reference.md:306]: I think @deepyaman is correct here. And to your question, it should be 2. without the article included in the hyperlink.

**Review by yetudada — COMMENTED** [review]
Thank you so much for this @SajidAlamQB! I agree with the changes that @deepyaman flagged.

File: docs/source/extend_kedro/common_use_cases.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ You can use [DataSets](/kedro.extras.datasets) to interface with various differe
 
 ## Use Case 3: How to add or modify CLI commands
 
-If you want to customise a built-in Kedro command, such as `kedro run`, for a specific project, add a `cli.py` file that defines a custom `run()` function. You should add the `cli.py` file at the same level as `settings.py`, which is usually the `src/PROJECT_NAME` directory. A template for the `cli.py` file can be found [here](../development/commands_reference.md#customise-or-override-project-specific-kedro-commands).
+If you want to customise a built-in Kedro command, such as `kedro run`, for a specific project, add a `cli.py` file that defines a custom `run()` function. You should add the `cli.py` file at the same level as `settings.py`, which is usually the `src/PROJECT_NAME` directory. See [the template for the `cli.py` file](../development/commands_reference.md#customise-or-override-project-specific-kedro-commands).
```
[the remark this answers is not in the document]
Comment (SajidAlamQB) [reply docs/source/extend_kedro/common_use_cases.md:19]: No worries, thank you for catching these out. 👍

File: docs/source/extend_kedro/common_use_cases.md [hunk 4]
```diff
@@ -31,12 +31,12 @@ def notebook_run(...):
     == ADD YOUR CUSTOM NOTEBOOK COMMAND CODE HERE ==
 ```
 
-To inject additional CLI commands intended to be reused across projects, please refer to our [plugins](./plugins.md) system. An example of one such command is the `kedro viz` command introduced by the official [Kedro-Viz](https://github.com/kedro-org/kedro-viz) plugin. This command is intended to work on every Kedro project and therefore must be a standalone plugin.
+To inject additional CLI commands intended to be reused across projects, please refer to [our plugins system](./plugins.md). An example of one such command is the `kedro viz` command introduced by the official [Kedro-Viz plugin](https://github.com/kedro-org/kedro-viz). This command is intended to work on every Kedro project and therefore must be a standalone plugin.
```
Comment (deepyaman) [inline docs/source/extend_kedro/common_use_cases.md:34]:
```suggestion
To inject additional CLI commands intended to be reused across projects, please refer to [our plugin system](./plugins.md). An example of one such command is the `kedro viz` command introduced by the [Kedro-Viz plugin](https://github.com/kedro-org/kedro-viz). This command is intended to work on every Kedro project and therefore must be a standalone plugin.
```
(or include "official" in the link name, as when talking about starters below)

File: docs/source/kedro_project_setup/configuration.md [hunk 5]
```diff
@@ -208,7 +208,7 @@ except MissingConfigException:
 The `kedro.framework.context.KedroContext` class uses the approach above to load project parameters.
 ```
 
-Parameters can then be used on their own or fed in as function inputs, as described [below](#use-parameters).
+Parameters can then be used on their own or fed in as function inputs, as described [below in Use Parameters](#use-parameters).
```
Comment (deepyaman) [inline docs/source/kedro_project_setup/configuration.md:211]: ~Parameters can then be used on their own or fed in as function inputs, as described below in the ["Use parameters" section](#use-parameters).~

Comment (deepyaman) [reply docs/source/kedro_project_setup/configuration.md:211]: Better yet IMO, and avoids weird quoting:
```suggestion
[Parameters can then be used on their own or fed in as function inputs.](#use-parameters)
```

File: docs/source/logging/experiment_tracking.md [hunk 6]
```diff
@@ -10,7 +10,7 @@ However, Kedro was missing a way to log metrics and capture all this logged data
 
 Experiment tracking in Kedro adds in the missing pieces and will be developed incrementally.
 
-The following section outlines the setup within your Kedro project to enable experiment tracking. You can also refer to [this tutorial](../tutorial/set_up_experiment_tracking.md) for a step-by-step process to access your tracking datasets on Kedro-Viz.
+The following section outlines the setup within your Kedro project to enable experiment tracking. You can also refer to the [set up experiment tracking tutorial](../tutorial/set_up_experiment_tracking.md) for a step-by-step process to access your tracking datasets on Kedro-Viz.
```
Comment (deepyaman) [inline docs/source/logging/experiment_tracking.md:13]:
```suggestion
The following section outlines the setup within your Kedro project to enable experiment tracking. You can also refer to the [set up experiment tracking tutorial](../tutorial/experiment_tracking.md) for a step-by-step process to access your tracking datasets on Kedro-Viz.
```
or I guess you could say "tutorial on setting up experiment tracking"

File: docs/source/tools_integration/pyspark.md [hunk 7]
```diff
@@ -126,7 +126,7 @@ assert isinstance(df, pyspark.sql.DataFrame)
 [Delta Lake](https://delta.io/) is an open-source project that enables building a Lakehouse architecture on top of data lakes. It provides ACID transactions and unifies streaming and batch data processing on top of existing data lakes, such as S3, ADLS, GCS, and HDFS.
 To setup PySpark with Delta Lake, have a look at [the recommendations in Delta Lake's documentation](https://docs.delta.io/latest/quick-start.html#python).
 
-We recommend the following workflow, which makes use of the [Transcoding](../data/data_catalog.md) feature in Kedro:
+We recommend the following workflow, which makes use of the [Transcoding feature in Kedro](../data/data_catalog.md):
```
Comment (deepyaman) [inline docs/source/tools_integration/pyspark.md:129]:
```suggestion
We recommend the following workflow, which makes use of the [transcoding feature in Kedro](../data/data_catalog.md):
```

**Review by deepyaman — COMMENTED** [review]
Thanks for making all those changes! I just took a quick pass through, but looks good for the most part.

**Review by deepyaman — APPROVED** [review]
🚀 🚀 🚀

**Pull request merged**